app/components/emailSignup/index.tsx

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react';
+import { LinkButton } from '../Button';
+import UserInput from '../common/userInput';
+import { Email } from '../icons';
+
+interface EmailSignupForm {
+  fullName: string;
+  email: string;
+  password: string;
+}
+
+const EmailSignup = () => {
+  const intialEmailFormState = {
+    fullName: '',
+    email: '',
+    password: '',
+  };
+  const [formState, setFormState] = useState<EmailSignupForm>(intialEmailFormState);
+
+  const updateFullName = (fullName: string) => {
+    setFormState((state) => ({ ...state, fullName }));
+  };
+
+  const updateEmail = (email: string) => {
+    setFormState((state) => ({ ...state, email }));
+  };
+
+  const updatePassword = (password: string) => {
```

Comment:
NIT: There seems to be some repetition of same code and functionality here. This is well written, however we could also think of having a config file that contains all details and map that to the UserInput component.
Again, we can revisit this in future

Reply:
> NIT: There seems to be some repetition of same code and functionality here. This is well written, however we could also think of having a config file that contains all details and map that to the UserInput component. Again, we can revisit this in future

agreed! We can have a discussion on how to approach this and make it modular.